indico/util/protection_info.py

```diff
@@ -0,0 +1,60 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2024 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from functools import cached_property
+
+from indico.modules.legal import legal_settings
+
+
+class UnprotectedPageInfo:
+    def __init__(self):
+        self.object = None
+        self.is_protected = False
+        self.access_restrictions = set()
+        self.networks_with_access = []
+        self.protection_mode = 'public'
+        self.disclaimer = ''
+
+
+class ProtectedPageInfo:
+    def __init__(self, protected_object):
+        self.object = protected_object
+
+    @cached_property
+    def is_protected(self):
+        return self.object.is_protected
+
+    @cached_property
+    def access_restirctions(self):
+        return self.object.get_access_list()
+
+    @cached_property
+    def networks_with_access(self):
+        return sorted(o.name for o in self.access_restirctions if o.principal_type.name == 'network')
+
+    @cached_property
+    def protection_mode(self):
+        if self.networks_with_access:
+            return 'network'
+        if self.is_protected:
+            return 'restricted'
+        return 'public'
+
+    @cached_property
+    def disclaimer(self):
+        if self.mode == 'network':
```

Comment:
`self.mode` does not exist

Reply:
Ah damn, yeah, I'll fix that when I rename "protection_mode".